lib/AST/ASTContext.cpp

```diff
@@ -1642,6 +1642,23 @@ void ASTContext::addModuleInterfaceChecker(
   getImpl().InterfaceChecker = std::move(checker);
 }
 
+void ASTContext::setModuleAliases(const llvm::StringMap<StringRef> &aliasMap) {
+  for (auto k: aliasMap.keys()) {
+    auto val = aliasMap.lookup(k);
+    if (!val.empty()) {
+      ModuleAliasMap[getIdentifier(k)] = getIdentifier(val);
```

Comment:
Does this (or something else) copy the strings storage to the AST context?

Reply:
I've had issues in the past with `StringRef`s parsed from arguments being lost in long running SourceKit process. I'm not sure if we could see the same thing here.

Reply:
`getIdentifier` allocates an entry so it won't get lost. Also the StringRefs in the argument map contains allocated data so they won't be lost either by the time this func is called. We could perhaps clear the argument map after the loop for better optimization if needed.

Reply:
Sounds good. Thanks for the details!